jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Dispatcher.java

```diff
@@ -188,7 +188,8 @@ protected void forward(ServletRequest request, ServletResponse response, Dispatc
                     query = old_uri.getQuery();
 
                 String decodedPathInContext = URIUtil.decodePath(_pathInContext);
-                baseRequest.onDispatch(HttpURI.build(old_uri, _uri.getPath(), _uri.getParam(), query), decodedPathInContext);
+                baseRequest.setHttpURI(HttpURI.build(old_uri, _uri.getPath(), _uri.getParam(), query));
+                baseRequest.setContext(_contextHandler.getServletContext(), decodedPathInContext);
```

Comment:
why not keep using `onDispatch` and within that method you can check the context is still correct?  Or maybe you need to pass in the context handler to that method?

Reply:
I was adjusting this to be what it was in Jetty-11 like you said.

In Jetty-11 there is no onDispatch method on the request but it has setHttpURI and setContext.